### DISCUSSION TITLE
~~appending with `a` while cursor at EOL should behave as `A`~~
While traversing words in a line with `w`, if `line-ending crlf`, selecting the last word with `w` causes the selection to include EOL. While `line-ending lf` doesn't cause that issue.

### DISCUSSION THREAD

**Discussion by masriomarm** [helix-editor/helix#7701] [Ideas] [answered]
while cursor at EOL, appending with `a` moves the cursor to the beginning of the next line.
recovering is a bit frustrating. I suggest while cursor is at EOL, appending wiht `a` should behave as appending to the end of line `A`.

**Comment by kirawi** [comment]
This is intentional behavior borrowed from Kakoune for the sake of consistency over special-casing EOL.

**Reply by masriomarm** [reply]
Hi @kirawi Thanks for your replay,
The issue seems to be with line ending, if `line-ending` is set to `crlf` using `w` to get the last word in line would cause the selection of the line ending character.
If `line-ending` is set to `lf`, selecting the last word in line with a moving `w` doesn't cause the issue and appending seems to append to the end of the word normally rather the beginning of the new line. Not sure if that's a bug or by decision, but it seems to be affecting windows since the crlf. I noticed the issue isn't present on my fedora.
However, setting `line-ending` to `lf` on Windows would  generate multiple diffs sadly.

**Reply by pascalkuthe** [reply, marked as answer]
the behaviour for `w` is indeed a bug (the classic using chars instead of graphemes) but the behaviour of `a` is very much intentional

**Reply by masriomarm** [reply]
Would you like me to edit the title and reference in a new issue?

**Reply by the-mikedavis** [reply]
Yep that sounds good to me 👍

**Reply by sicher** [reply]
Does #8111 fix the `w` behaviour bug?

**Reply by pascalkuthe** [reply]
no that PR is about commands like `f` and `t`. `w` is a completely unrelated code that does not share the logic here (the cause of the bug is also a different one)
